absl/strings/escaping_test.cc

```diff
@@ -278,6 +278,98 @@ TEST_F(CUnescapeTest, UnescapesMultipleUnicodeNulls) {
                    "\0", 5), result_string_);
 }
 
+TEST(CUnescape, NonNullTerminatedOctal1) {
+  absl::string_view reference("\\111", 2);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1");
+}
+
+TEST(CUnescape, NonNullTerminatedOctal2) {
+  absl::string_view reference("\\1z1", 3);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1z");
+}
+
+TEST(CUnescape, NonNullTerminatedHex) {
+  absl::string_view reference("\\x11", 3);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\x1");
+}
+
+TEST(CUnescape, NonNullTerminatedInvalidHex) {
+  absl::string_view reference("\\x1", 2);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(CUnescape, NonNullTerminatedInvalidUtf1) {
+  absl::string_view reference("\\u11111", 5);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(CUnescape, NonNullTerminatedInvalidUtf2) {
+  absl::string_view reference("\\u1z11", 6);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(CUnescape, NonNullTerminatedInvalidUtf3) {
+  absl::string_view reference("\\U0000000000000", 9);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(CUnescape, InvalidHex) {
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape("\\xz", &unescaped_str));
+}
+
+TEST(CUnescape, InvalidUtf) {
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape("\\U1z111111", &unescaped_str));
+}
+
+TEST(CUnescape, InvalidEscaping) {
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape("\\z", &unescaped_str));
+}
+
+TEST(CUnescape, OctalLimits) {
+  std::string reference = "\\377";
+  std::string expected = "\377";
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+  EXPECT_EQ(unescaped_str, expected);
+
+  EXPECT_FALSE(absl::CUnescape("\\400", &unescaped_str));
+}
+
+TEST(CUnescape, UtfULimits) {
```

Comment:
This is really about Unicode rather than UTF, so how about calling this `UnicodeLimits`?